src/internal/response_buffer.go

```diff
@@ -0,0 +1,71 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"bytes"
+	"net/http"
+)
+
+// ResponseBuffer is a wrapper for the http.ResponseWriter to buffer the response data
+type ResponseBuffer struct {
```

Comment:
why not use httputil.ResponseRecorder

Reply:
+1

Reply:
You mean `httptest.ResponseRecorder`? It doesn't meet our requirement.
The `ResponseBuffer` wraps an existing `ResponseWriter` but the `httptest.ResponseRecorder` doesn't

Reply:
I don't disagree we replace the `httptest.ResponseRecorder`.

But, if you are trying to use the `ResponseBuffer` to replace `responseWriter` there may be a problem.  There are a few contract in responseWriter for example the `WriteHeader` can be called only once.  But you are overwriter the status code.

More details see the implementation of httptest:
https://golang.org/src/net/http/httptest/recorder.go#L125

Reply:
Done